util.go

```diff
@@ -22,6 +23,11 @@ func until(f func(), period time.Duration, stopCh <-chan struct{}) {
 		case <-stopCh:
 			return
 		case <-time.After(period):
+			if period*2 <= maxPeriod {
+				period *= 2
+			} else {
+				period = initialPeriod
+			}
```

Comment:
this seems.... odd to me.  Am I reading this right?  This seems like it would wait:

1s, 2s, 4s, 8s, 16s, 1s, 2s, 4s, 8s, 16s, 1s, 2s, 4s, 8s, 16s....

Reply:
Generally when I write things like this, what you want is some signal of success or failure from `f()`.  As long as f errors you backoff from initial -> max.  Whenever f succeeds you reset back to initial.  A reasonable signal in this particular case might be "If I got an EOF that was a successful connection that terminated eventually; if I got a different kind of error that was a failure to connect"